src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -11735,6 +11735,10 @@ void Unit::Dismount()
 
     if (Player* thisPlayer = ToPlayer())
         thisPlayer->SendMovementSetCollisionHeight(thisPlayer->GetCollisionHeight(false));
+	
+	// for dismount inside Vashj'ir
+	if(ToPlayer()->HasAura(86510))
+		ToPlayer()->RemoveAura(86510);
 
```

Comment:
this is place for crash.. ToPlayer is not valid pointer.
better is

if (Player* thisPlayer = ToPlayer())
	{
		thisPlayer->SendMovementSetCollisionHeight(thisPlayer->GetCollisionHeight(false));
 	
		// for dismount inside Vashj'ir
		if(thisPlayer->HasAura(86510))
			thisPlayer->RemoveAura(86510);
	}